### DISCUSSION TITLE
What if waved server is packaged with pypi uploads?

### DISCUSSION THREAD

**Discussion by sairaamv** [h2oai/wave#1124] [Ideas] [answered]
@lo5 , @mturoci , Is it feasible to pack waved server with pypi upload?
If the user executes `wave run app` , can we spin off waved on the backend and hookup the app as well?

Would like to understand the challenges with this feature request. Can PR and take a stab at it but but would like to understand/discuss the challenges/caveats. 
If it could be taken further, what should be considered? 
If it is not, why?

Thanks

**Comment by lo5** [comment, marked as answer]
We want to do this. It'll definitely make the getting started experience much simpler. 

This is mainly packaging work:
- Modify setuptools / `setup.py` and test against PyPI (90% of the work)
- Make `wave run` automatically launch `waved` (using `subprocess` or `POpen`).
 
Today we push one single platform-neutral `none-any` `whl` file to PyPI. Instead, we need to push at least 3 platform-specific whl files - `x86_64` for windows, osx and linux. We already have makefile tasks that produce `waved` for windows, osx and linux - it's just a matter of building them first and copying them into the whl build directory before invoking `setup.py bdist_wheel`, so that they're available inside the `whl` file. Once pip-installed, `wave run` should be able to run `waved` using a relative path.

**Comment by mturoci** [comment]
@sairaamv this is now implemented and released in nightly builds. Feel free to check it out. Simply choose your os-specific wheel, install it, hit `wave run your-app` and you should have both waved and your app running.

**Reply by sairaamv** [reply]
This is awesome!
